expat/configure.ac

```diff
@@ -183,33 +183,75 @@ AC_LINK_IFELSE([AC_LANG_SOURCE([
       AC_MSG_RESULT([yes])],
      [AC_MSG_RESULT([no])])])
 
-
-AC_MSG_CHECKING([for getrandom (Linux 3.17+, glibc 2.25+)])
-AC_LINK_IFELSE([AC_LANG_SOURCE([
-  #include <stdlib.h>  /* for NULL */
-  #include <sys/random.h>
-  int main() {
-    return getrandom(NULL, 0U, 0U);
-  }
-  ])],
-  [AC_DEFINE([HAVE_GETRANDOM], [1], [Define to 1 if you have the `getrandom' function.])
-   AC_MSG_RESULT([yes])],
-  [AC_MSG_RESULT([no])
-
-   AC_MSG_CHECKING([for syscall SYS_getrandom (Linux 3.17+)])
+AC_ARG_WITH([usegetrandom],
+ [AS_HELP_STRING([--with-usegetrandom],
+ [Use getrandom function in the system @<:@default=check@:>@])],
+ [],
+ [with_usegetrandom=check])
+          
+AS_IF([test "x$with_usegetrandom" != xno],
+  [AC_MSG_CHECKING([for getrandom (Linux 3.17+, glibc 2.25+)])
+   AC_LINK_IFELSE([AC_LANG_SOURCE([
+   #include <stdlib.h>  /* for NULL */
+   #include <sys/random.h>
+   int main() {
+     return getrandom(NULL, 0U, 0U);
+   }
+   ])],
+   [
+	AC_DEFINE([HAVE_GETRANDOM], [1], [Define to 1 if you have the `getrandom' function.])
+    AC_MSG_RESULT([yes])
+    getrandom_detect=yes
+   ],
+    [AC_MSG_RESULT([no])]
+     getrandom_detect=no)
```

Comment:
Use of whitespace is rather inconsistent in the added code, including mixed tabs and spaces but not only. Please find some formatting that is at least consistent to itself and to some other parts of the file.